lambdas/lambda_edge.py

```
@@ -1,10 +1,107 @@
+    if jwt == "":
+        return False
+
+    decoded = decode_jwt(jwt, jwt_secret)
+    if decoded == "invalid":
+        return "invalid"
+
+    if decoded == "expired":
+        return "expired"
+
+    # valid JWT: user is legit for general access. This is where
+    # apps needing specific user auth and not general user auth would
+    # need to check the user that was returned is authorized, not just
+    # authenticated. For our initial purposes, authenticated is all we
+    # will be checking for
+    return "valid"
+
+
+def decode_jwt(usertoken, jwt_secret):
+    try:
+        jwt.decode(usertoken, jwt_secret, algorithms=["HS256"])
+    except jwt.ExpiredSignatureError:
+        # Signature has expired
+        logging.exception("JWT token error: jwt.ExpiredSignatureError")
+        return "expired"
+    except jwt.InvalidSignatureError:
+        # Signature is invaid
+        logging.exception("JWT token error: jwt.InvalidSignatureError")
+        return "invalid"
+    except jwt.DecodeError:
+        # Bogus JWT token
+        logging.exception("JWT token error: jwt.DecodeError")
+        return "invalid"
+    return "valid"
+
+
+def handler(event, _context):
+    request = event["Records"][0]["cf"]["request"]
+    headers = request["headers"]
+
+    """
+    Check for session-id in request cookie in viewer-request event,
+    if session-id is absent, redirect the user to sign in page with original
+    request sent as redirect_url in query params.
+    """
+
+    client = boto3.client("ssm", region_name="us-east-1")
+
+    ssm_params = client.get_parameters_by_path(
+        Path="/apps/cf-lambda-geo-auth/", WithDecryption=False
+    )
+
+    logging.info(ssm_params)
+
+    for param in ssm_params["Parameters"]:
+        if param["Name"] == "/apps/cf-lambda-geo-auth/jwt-secret":
+            jwt_secret = param["Value"]
+        if param["Name"] == "/apps/cf-lambda-geo-auth/auth-url":
+            auth_url = param["Value"]
+
```

Wonder if we should throw an exception here if not all the expected SSM params are found?  

The advantage I see would be easy debugging if the lambda and TF code got out of sync, where otherwise we'd just get errors about undefined variables.